### PR TITLE
Setup Cloudinary #21

### PR DESCRIPTION
**Pull request by TawfiqKhan**
### What this PR does (required):
- Add a field to upload avatar (uploading avatar is optional)
- Add route for updating user avatar
- Add route for uploading images for the card
- Add route for updating images when editing card

### Screenshots / Videos (required):
E717AD68-9B48-4AC5-B0DA-8F1AE423931C

### Any information needed to test this feature (required):
- Uncomment codes in the dashboard.tsx and app.tsx. Current code contains a hardcoded dashboard and signup/login doesn't work with those codes present.

### CODE REVIEW

**Review by ajaytomgeorge — COMMENTED** [review]
@TawfiqKhan  Great work . Some comments

File: client/src/components/ImageForm.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  };
+
+  return (
+    <div>
+      <form action="">
+        <input onChange={handleChange} type="text" name="text" value={text} />
+        <input onChange={fileSelected} type="file" accept="image/*"></input>
+        <button onClick={submitForm}>submit</button>
+      </form>
+      <img src={`https://water-polo-hatchways.s3.ca-central-1.amazonaws.com/5a7cd3c023db2417a00f78e54e27f566`} alt="" />
```
[the remark this answers is not in the document]
Comment (TawfiqKhan) [reply client/src/components/ImageForm.tsx:34]: that was just a test component. which I removed in a later commit,

File: server/controllers/auth.js [hunk 2]
```diff
@@ -1,7 +1,11 @@
 const User = require("../models/User");
-const Board = require("../models/Board");
-const Column = require("../models/Column");
-const Card = require("../models/Card");
+// const Board = require("../models/Board");
+// const Column = require("../models/Column");
+// const Card = require("../models/Card");
```
[the remark this answers is not in the document]
Comment (TawfiqKhan) [reply server/controllers/auth.js:4]: removed!

File: server/controllers/image.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+const multer  = require('multer')
+// const upload = multer({ dest: 'uploads/' })
```
[the remark this answers is not in the document]
Comment (TawfiqKhan) [reply server/controllers/image.js:2]: removed thanks

File: server/controllers/card.js [hunk 4]
```diff
@@ -16,20 +28,42 @@ exports.createCard = asyncHandler(async (req, res, next)=> {
...
+  if (!card) {
+    res.status(404);
+    throw new Error("No Card found");
+  }
+
+  const {currentImages, title, tag} = req.body
+  // currentImages is an array containing the keys of current images in the card
+  // update the images in db based on the current images in the card
+  if(currentImages.length) {
+    await deleteImagesInS3(card, currentImages)
```
[the remark this answers is not in the document]
Comment (TawfiqKhan) [reply server/controllers/card.js]: I initially had it in the file but later deleted it by mistake, I have moved it to the utils folder now.

**Review by moffatethan — APPROVED** [review]

**Pull request merged**